### ISSUE TITLE
✨[Improvement] Unified + Selective Decomposition System for Dynamo

### ISSUE THREAD

**Issue by gs-olive** [pytorch/TensorRT#2160]
# Context
Currently, the decomposition system in compile, soon to be integrated into export, is based on the [`core_aten_decompositions`](https://github.com/pytorch/pytorch/blob/6d2162e644152efae91d22208984f1dd2f9bd999/torch/_decomp/__init__.py#L180) set, as here:
https://github.com/pytorch/TensorRT/blob/f7b03f466f62c1027f4025b60a4f5e1f854550a6/py/torch_tensorrt/dynamo/lowering/_decompositions.py#L5
This set of decompositions, while exhaustive, decomposes certain operations which we would like to preserve, including `aten.cudnn_batch_norm` and other batch norm implementations. Such implementations, when combined with utilities for freezing, such as those discussed in #2124, #2128, can greatly improve performance on networks with these operators.

As a bonus, the pre-aot decompositions, [as here](https://github.com/pytorch/TensorRT/tree/f7b03f466f62c1027f4025b60a4f5e1f854550a6/py/torch_tensorrt/dynamo/lowering/substitutions), may be more easily removed/reduced, since it is difficult to integrate these into export currently.

# Proposal
Introduce a new decomposition dictionary which does one of two things:

  **1.** Selectively _remove_ operators from the `core_aten_decompositions` dictionary, to ensure these operators are not decomposed and will stay in their initial state. This allows us to continue using the most up-to-date decompositions dictionary while selectively removing the operators which are preferable to stay as-is.
  **2.** Selectively _keep_ operators in the `core_aten_decompositions` dictionary, to ensure the set of decompositions is fully defined and does not depend on the state of PyTorch's dictionary. This gives more control over the set of decompositions allowed, but requires that modifications are made to the decompositions library more often, since PyTorch may introduce new desirable decompositions often.
